client/src/common/client.ts

```diff
@@ -276,7 +277,7 @@ class DefaultErrorHandler implements ErrorHandler {
 
 	private readonly restarts: number[];
 
-	constructor(private name: string) {
+	constructor(private name: string, private restartTimes: number) {
```

Comment:
I would suggest to name this maxRestartCount. We usually use count instead of times for this.

Reply:
Done.